### PR TITLE
chore(pp-upgrade): test functionality after provisionPool and bank upgrade

### LINKED ISSUE #10395: Contract Upgrade: V28 ProvisionPool and V14 Bank
## What is the Problem Being Solved?

As mentioned in #8158, we need the ability to restart or replace all vats. This issue is focused on upgrading **v28-ProvisionPool** and **v14-bank**. They are related because of #8722 and #8724, which were addressed by #9481.

## Description of the Design

These upgrades have been tested previously, and are in `agoric-3-proposals/upgrade-17`. The goal here is to reinstate those in `a3p-integration`, and stage them in `upgrade.go` for the next upgrade.

In addition, `vat-bank` is a client of `feeDistributor`, which will be [separately be upgraded](https://github.com/Agoric/agoric-sdk/issues/10393). Let's re-review the tests to be sure that vat-bank continues to send fees after an upgrade.

## Security Considerations

vat-bank is a `critical` vat. provisionPool is involved in creating new smartWallets.

## Scaling Considerations

not significant.

## Test Plan

Move `proposals/78:upgrade-17/provisionPool.test.js` from `agoric-3-proposals` to `a3p-integration`. Verify that it checks everything relevant.

## Upgrade Considerations

See above.

### LINKED ISSUE #10425: convert `makeBridgeProvisionTool` from a `Far` to an `Exo`
## What is the Problem Being Solved?
https://github.com/Agoric/agoric-sdk/pull/10419#issuecomment-2463441008 talks about converting the maker function of the bridge handler created in `provisionPoolKit.js` from a `Far` object to an `Exo` object t make it durable. This is needed because https://github.com/Agoric/agoric-sdk/issues/8849 shows `makeBridgeProvisionTool` returning a `Far` (along with other objects) prevents the bootstrap vat from upgrading. 

## Description of the Design
```js
const prepareBridgeProvisionTool = (baggage) => {
  const makeHandler = prepareExoClass(
    baggage,
    'provisionHandler',
    BridgeHandlerI,
    (sendInitialPayment, onProvisioned) => ({ sendInitialPayment, onProvisioned }),
    {
      fromBridge: async obj => {
          obj.type === 'PLEASE_PROVISION' ||
            Fail`Unrecognized request ${obj.type}`;
          trace('PLEASE_PROVISION', obj);
          const { address, powerFlags } = obj;
          powerFlags.includes(PowerFlags.SMART_WALLET) ||
            Fail`missing SMART_WALLET in powerFlags`;

          const bank = E(bankManager).getBankForAddress(address);
          // only proceed if we can provide funds
          await sendInitialPayment(bank);

          const [_, created] = await E(walletFactory).provideSmartWallet(
            address,
            bank,
            namesByAddressAdmin,
          );
          if (created) {
            onProvisioned();
          }
          trace(created ? 'provisioned' : 're-provisioned', address);
        }, 
    }
  );
  
  return makeHandler;
}
```

### Open Question: Are we sure `handler` is assigned to `provisionWalletBridgeManager`?
https://github.com/Agoric/agoric-sdk/blob/4b1403aa06a6c08aae2d4cce0fd582480b7e3c7f/packages/vats/src/core/chain-behaviors.js#L152-L186

In the snippet above the `handler` seems to be assigned to a `Far` assuming `provisioning` is not `undefined`. `makeProvisioner` looks like it is assigning `provisioning` to `vat-provisioning`.

https://github.com/Agoric/agoric-sdk/blob/4b1403aa06a6c08aae2d4cce0fd582480b7e3c7f/packages/vats/src/core/chain-behaviors.js#L116-L125

My concern is that even if we make `makeBridgeProvisionTool` an `Exo`, I am not so sure it will help us make the `handler` vat-bootstrap holding on durable. Could you please help me understand this? @Chris-Hibbert @dckc 

## Test Plan
Need to make sure both auto-provisioning and `agd tx swingset provision-one` works in `a3p`.

### LINKED ISSUE #10564: reduce complexity by dropping `setReferences` and start using `privateArgs` instead in `provisionPool`
## What is the Problem Being Solved?
Currently the `setReferences` method  is used to pass `bankManager`, `walletFactory` and `namesByAddressAdmin` into the `provisionPool`. This creates complexity when trying to instantiate `exo` objects that depend on these powers.

The problem occurs during an upgrade. SwingSet expects all `exo` objects defined in the previous incarnation to be instantiated during the start up. If there's an `exo` object that depends on state that is set after the startup, like calling `setReferences` after the start up is complete, the work around causes code get more complex.

## Description of the Design
Move `bankManager`, `walletFactory` and `namesByAddressAdmin` into `privateArgs`

### PR DESCRIPTION
**Pull request by anilhelvaci**
closes: #10395
closes: #10425 
closes: #10564

## Description

As mentioned in https://github.com/Agoric/agoric-sdk/issues/8158, we need the ability to restart or replace all vats. This PR focuses on the vats v28-provisionPool and v14-bank. The goal is to make sure after upgrading both of those vats, v28-provisionPool functionality keeps working. We pay special attention to https://github.com/Agoric/agoric-sdk/issues/8722 and https://github.com/Agoric/agoric-sdk/issues/8724 during the tests as those issues were identified as potential problems against v28-provisionPool upgrade.

### Security Considerations

v28-provisionPool and v14-bank are critical vats for the system when it comes to onboarding new users and keeping track of ERTP representations of user assets. Reviewers, please highlight the slightest risk you see.

### Scaling Considerations

v28-provisionPool vat has a linear relationship with the number of users entering the Agoric system. So it is pretty important it keeps working. Reviewers, please highlight the slightest risk you see.

### Documentation Considerations

None.

### Testing Considerations

During the testing, we focused on `provisionPoolAddress`' cosmos level balances as our source of truth. The reasoning behind this is; if the IST balance of the `provisionPoolAddress` it can only mean
- it has received the anchor asset we've sent 
- v28-provisionPool is notified of this balance change
- executed a swap against corresponding PSM
- deposited the payout to IST purse
- v14-bank updated the balances correctly

In our test we send two anchor assets;
* USDC_axl to make sure v28-provisionPool recovered existing purses
* USD_LEMONS to make sure v28-provisionPool is notified of new assets

### Upgrade Considerations

Both v28-provisionPool and v14-bank are staged upgrades in `upgrade.go`

### CODE REVIEW

File: golang/cosmos/app/upgrade.go
Comment (Chris-Hibbert) [inline golang/cosmos/app/upgrade.go]: Please put this in a separate section with a comment saying it's for upgrade 19 as is done in #10418.

File: a3p-integration/proposals/p:upgrade-19/test-lib/sync-tools.js
Comment (Chris-Hibbert) [inline a3p-integration/proposals/p:upgrade-19/test-lib/sync-tools.js:1]: Duplicating these files is the wrong solution. AIUI, the problem is that when docker runs the proposals, it only mounts a single proposal's directory at a time. I've talked this over with @turadg, and I'd like to try using symbolic links to manage this.

Please add a `lib` directory in `a3p-integration` next to `proposals`, and put a canonical copy of the file there. Then make each of the proposals that want to import all or some of `sync-tools` include a symbolic link to that file, so the tests can import `./sync-tools.js`. Do the same with errors.js.

I haven't verified how this will work with github, but I'd like to try the approach out here.

File: a3p-integration/proposals/p:upgrade-19/provisionPool.test.js
Comment (Chris-Hibbert) [inline a3p-integration/proposals/p:upgrade-19/provisionPool.test.js]:
```suggestion
    { errorMessage: 'Provision pool not able to swap USDC_axl for IST.' },
```

File: a3p-integration/proposals/p:upgrade-19/provisionPool.test.js
Comment (Chris-Hibbert) [inline a3p-integration/proposals/p:upgrade-19/provisionPool.test.js]: Please check the incarnation #s of both vats. I think the tests would pass if the upgrade didn't take place since there's no change in behavior to detect.

**Review by Chris-Hibbert — CHANGES_REQUESTED** [review]
I suspect that it would be a good idea to convert `makeBridgeProvisionTool` from a *Far* object to an Exo, but I'm not positive. I've asked for advice and will let you know what I hear.

**Comment by Chris-Hibbert** [comment]
> I suspect that it would be a good idea to convert makeBridgeProvisionTool from a Far object to an Exo, but I'm not positive. I've asked for advice and will let you know what I hear.

The response I got was "yes, convert it to an Exo", but it appears that this object may be one of those discussed in #8849, which may mean we need a clearer story on upgrading the bootstrap vat (V1) before we can proceed to upgrade provisionPool.

You can work on converting it to an Exo

I need to have more design discussions with the kernel team about the impact. The concern seems to be about whether auto-provisioning will continue to work.  The links mentioned were 
 *   https://community.agoric.com/t/proposal-65-agoric-upgrade-13-smart-wallet-auto-provisioning-interchain-stack/560
 *  do a PLEASE_PROVISION, see [agd tx swingset provision-one docs](https://docs.agoric.com/guides/agoric-cli/agd-query-tx.html#agd-tx-swingset-provision-one)
 *   for auto-provision, try making a vault from an account that has never done a PLEASE_PROVISION

**Comment by anilhelvaci** [comment]
> The response I got was "yes, convert it to an Exo", but it appears that this object may be one of those discussed in #8849, which may mean we need a clearer story on upgrading the bootstrap vat (V1) before we can proceed to upgrade provisionPool.
> 
> You can work on converting it to an Exo
> 
> I need to have more design discussions with the kernel team about the impact. The concern seems to be about whether auto-provisioning will continue to work. The links mentioned were
> 
> * https://community.agoric.com/t/proposal-65-agoric-upgrade-13-smart-wallet-auto-provisioning-interchain-stack/560
> * do a PLEASE_PROVISION, see [agd tx swingset provision-one docs](https://docs.agoric.com/guides/agoric-cli/agd-query-tx.html#agd-tx-swingset-provision-one)
> * for auto-provision, try making a vault from an account that has never done a PLEASE_PROVISION

#10425 tracks this now. From your comment I understand the path for this PR to land;
* Close 10425
* wait the plan for upgrading `vat-bootstrap` is clear
* address change requests

Could you please confirm? @Chris-Hibbert

**Comment by Chris-Hibbert** [comment]
> https://github.com/Agoric/agoric-sdk/issues/10425 tracks this now. From your comment I understand the path for this PR to land;

> * Close 10425
> * wait the plan for upgrading `vat-bootstrap` is clear
> * address change requests

> Could you please confirm? 

Let's pursue 10425. If we can write tests that show the bootstrap vat doesn't break, and we can continue to provision new wallets, I don't think we have to be able to upgrade vat-bootstrap before upgrading provisionPool.

File: a3p-integration/proposals/p:upgrade-19/provisionPool.test.js
Comment (anilhelvaci) [reply a3p-integration/proposals/p:upgrade-19/provisionPool.test.js]: resolved in https://github.com/Agoric/agoric-sdk/commit/bc5a94ea6fe566b490104abd7773b0ad614a1c2d

File: a3p-integration/proposals/p:upgrade-19/test-lib/sync-tools.js
Comment (anilhelvaci) [reply a3p-integration/proposals/p:upgrade-19/test-lib/sync-tools.js:1]: started using `@agoric/client-utils` in https://github.com/Agoric/agoric-sdk/commit/bc5a94ea6fe566b490104abd7773b0ad614a1c2d

File: a3p-integration/proposals/p:upgrade-19/.gitignore
Comment (Chris-Hibbert) [inline a3p-integration/proposals/p:upgrade-19/.gitignore]: please add a final linefeed

File: a3p-integration/proposals/p:upgrade-19/nullUpgradePP/null-upgrade-pp-permit.json
Comment (Chris-Hibbert) [inline a3p-integration/proposals/p:upgrade-19/nullUpgradePP/null-upgrade-pp-permit.json]: please add a final linefeed

File: a3p-integration/proposals/p:upgrade-19/depositUSD-LEMONS/deposit-usd-lemons.js
Comment (Chris-Hibbert) [inline a3p-integration/proposals/p:upgrade-19/depositUSD-LEMONS/deposit-usd-lemons.js:1]: I think the files in `depositUSD-LEMONS` and `nullUpgradePP` shouldn't have been included

File: golang/cosmos/app/upgrade.go
Comment (Chris-Hibbert) [inline golang/cosmos/app/upgrade.go]: shouldn't this be in the commented-out section?

File: packages/inter-protocol/src/provisionPoolKit.js
Comment (Chris-Hibbert) [inline packages/inter-protocol/src/provisionPoolKit.js]: The old code called `makeBridgeProvisionTool()` on every call of `makeHandler()`. Why does this switch to calling it only  once per incarnation of the ProvisionPool?

I don't think we need ephemera at all. (I might be wrong. Tell me why you think we want it.)

`makeHandler()` isn't called very often (only when `smartWallet` starts up, I think), so it's fine to create these on demand. 

I don't see the value in using `ephemera` to cache the handlers. Was there some other reason to use `ephemera`?

File: packages/inter-protocol/src/provisionPoolKit.js
Comment (Chris-Hibbert) [inline packages/inter-protocol/src/provisionPoolKit.js]: If you intend to leave this trace enabled, please add a few words about what's being printed. I think it would be fine to drop it once it's working.

File: packages/inter-protocol/src/provisionPoolKit.js
Comment (Chris-Hibbert) [inline packages/inter-protocol/src/provisionPoolKit.js]: I think `callWhen` is only useful when there's a parameter to await. see [the endo docs](https://github.com/endojs/endo/blob/c6e795e5431d5a2afc644a683bb2ef2690a694a5/packages/patterns/src/types.js#L470).

`calllWhen(m.await(foo))` says to call the method (in this case `makeHandler`) after waiting for `foo` to resolve, and then passing the resolved value to the method. This allows the method to be written so that it presumes the value has resolved, while allowing the caller to provide a promise.

**Review by Chris-Hibbert — CHANGES_REQUESTED** [review]

File: a3p-integration/proposals/p:upgrade-19/.gitignore
Comment (anilhelvaci) [reply a3p-integration/proposals/p:upgrade-19/.gitignore]: will do

File: a3p-integration/proposals/p:upgrade-19/nullUpgradePP/null-upgrade-pp-permit.json
Comment (anilhelvaci) [reply a3p-integration/proposals/p:upgrade-19/nullUpgradePP/null-upgrade-pp-permit.json]: will do

File: a3p-integration/proposals/p:upgrade-19/depositUSD-LEMONS/deposit-usd-lemons.js
Comment (anilhelvaci) [reply a3p-integration/proposals/p:upgrade-19/depositUSD-LEMONS/deposit-usd-lemons.js:1]: I think they should be because those file are nor auto generated like other submissions.

File: golang/cosmos/app/upgrade.go
Comment (anilhelvaci) [reply golang/cosmos/app/upgrade.go]: Oh sorry. I'll fix it.

File: packages/inter-protocol/src/provisionPoolKit.js
Comment (anilhelvaci) [reply packages/inter-protocol/src/provisionPoolKit.js]: Sure, will drop it

File: packages/inter-protocol/src/provisionPoolKit.js
Comment (anilhelvaci) [reply packages/inter-protocol/src/provisionPoolKit.js]: > The old code called makeBridgeProvisionTool() on every call of makeHandler(). Why does this switch to calling it only once per incarnation of the ProvisionPool?

Could you please take another look? `makeBridgeProvisionTool` is the maker returned from `prepareBridgeProvisionPool` so as far as I can see we are invoking the maker every time there's a call to `makeHandler`.

> I don't think we need ephemera at all. (I might be wrong. Tell me why you think we want it.)

> I don't see the value in using ephemera to cache the handlers. Was there some other reason to use ephemera?

The reason why we use ephemera is because of the `helper.sendInitialPayment` and `helper.onProvisioned` callbacks. We have to pass them like `prepareBridgeProvisionTool(zone, helper.sendInitialPayment, helper.onProvisioned)` because these specific functions are not durable(the `helper` facet is, but specific callbacks are not).  We create the maker in `helper.start` and create the ephemera like below.

https://github.com/Agoric/agoric-sdk/blob/a8b124a78d467f678caa903bf9b71ef00a13f76f/packages/inter-protocol/src/provisionPoolKit.js#L480-L487

If we try to pass those callbacks like:
```js
makeBridgeProvisionTool(
  bankManager, 
  walletFactory, 
  namesByAddressAdmin, 
  helper.sendInitialPayment, 
  helper.onProvisioned
)
```
Wwe get an error like `Remotables must be explicitly declared`.
 > Also tried `makeBridgeProvisionTool(bankManager, walletFactory, namesByAddressAdmin, bank => helper.sendInitialPayment(bank), () => helper.onProvisioned())`

I also tried;
```js
makeBridgeProvisionTool(
  bankManager, 
  walletFactory, 
  namesByAddressAdmin, 
  Far('helpers', { 
    sendInitialPayment: bank => helper.sendInitialPayment(bank), 
    onProvisioned: () => helper.onProvisioned()
  }
)
```

This time the exo package could serialize `Far('helpers')` but complained that it's not durable.

So the only solution I could think of was to;
* Define (call `prepareBridgeProvisioningTool`) bridge handler exo in `helper.start`
  * Make sure `prepareBridgeProvisioningTool` is fed with `helper.sendInitialPayment` and `helper.onProvisioned`
* In `helper.start` set the maker into the ephemera like: `ephemera.makeBridgeProvisionTool = makeBridgeProvisionTool;`
* In `makeHandler` consume references (written by `setReferences`) and `ephemera.makeBridgeProvisionTool`
* Use `ephemera.makeBridgeProvisionTool` to _instantiate_ the handler exo

### Why not define the exo in `makeHandler`?
SwingSet needs every exo object defined in the previous incarnation to be defined again during the start up phase. So defining it in `makeHandler` does not work since it is not invoked during the startup (needs `setReferences` to be called first).

### Why ephemera?
Because the properties of `this.state` in the `provisionPoolKit` are immutably defined during the first incarnation. So it is not possible to assign the maker for bridgeProvisioningHandler to the exo state like `this.state.makeBridgeProvisionHandler = makeBridgeProvisionHandler`. As a result we use ephemera to keep the reference to `makeBridgeProvisionHandler`.

Comment (Chris-Hibbert) [reply packages/inter-protocol/src/provisionPoolKit.js]: Yeah, you can't pass methods of a durable object, but you can pass a facet, and `helper` is just such a facet. Change the signature for `prepareBridgeProvisionTool` to take a `helper` rather than two functions. Actually, `helper` has 7 methods and `prepareBridgeProvisionTool` only needs two. Add a new facet purely for its use:

```
forHandler: {
  sendInitialPayment(destBank) {
               return this.facets.helper.sendInitialPayment(destBank);
  },
  onProvisioned() {
               return this.facets.helper.onProvisioned();
  },
}
```

and then the call to `prepareBridgeProvisionTool` looks like 
```
prepareBridgeProvisionTool(
            handlerZone,
            this.facets.forHandler,
          );
```

File: a3p-integration/proposals/p:upgrade-19/.gitignore
Comment (anilhelvaci) [reply a3p-integration/proposals/p:upgrade-19/.gitignore]: resolved in https://github.com/Agoric/agoric-sdk/commit/a7d00fc6fcfa78e7b88d42e370670d647652c4f6

File: a3p-integration/proposals/p:upgrade-19/nullUpgradePP/null-upgrade-pp-permit.json
Comment (anilhelvaci) [reply a3p-integration/proposals/p:upgrade-19/nullUpgradePP/null-upgrade-pp-permit.json]: resolved in https://github.com/Agoric/agoric-sdk/commit/a7d00fc6fcfa78e7b88d42e370670d647652c4f6

File: packages/inter-protocol/src/provisionPoolKit.js
Comment (anilhelvaci) [reply packages/inter-protocol/src/provisionPoolKit.js]: resolved in https://github.com/Agoric/agoric-sdk/commit/a7d00fc6fcfa78e7b88d42e370670d647652c4f6

File: packages/inter-protocol/src/provisionPoolKit.js
Comment (anilhelvaci) [reply packages/inter-protocol/src/provisionPoolKit.js]: resolved in https://github.com/Agoric/agoric-sdk/commit/a7d00fc6fcfa78e7b88d42e370670d647652c4f6

File: packages/inter-protocol/src/provisionPoolKit.js
Comment (Chris-Hibbert) [inline packages/inter-protocol/src/provisionPoolKit.js:320]: good catch!  I thought I saw that one of these was being used elsewhere, and so needed to be kept in `helper`, but all the uses are for the handler.

File: packages/inter-protocol/src/provisionPoolKit.js
Comment (Chris-Hibbert) [inline packages/inter-protocol/src/provisionPoolKit.js]: Renaming the `forHandler` facet to be called `helperFacet` here is confusing.
```suggestion
    (bankManager, walletFactory, namesByAddressAdmin, forHandler) => ({
```

File: a3p-integration/proposals/p:upgrade-19/provisionPool.test.js
Comment (Chris-Hibbert) [inline a3p-integration/proposals/p:upgrade-19/provisionPool.test.js]: please pull `bankSend`, `provision`, `introduceAndProvision`, `getProvisionedAddress`,  and `checkUserProvisioned` out to a separate library file that can be shared. We shouldn't  have to define these more than once.

Maybe `getProvisionedAddress` is simple enough that it doesn't have to go in the library.

File: a3p-integration/proposals/p:upgrade-19/provisionPool.test.js
Comment (Chris-Hibbert) [inline a3p-integration/proposals/p:upgrade-19/provisionPool.test.js]: In a3p and mainNet, this is v28, but that might not always be true. `getDetailsMatchingVats()` returns `name` as the exact name of the vat, so you can validate by comparing the name using `===`.

 I didn't fix `getVatDetails`, so that one returns the `name` provided as an argument, and wouldn't distinguish "provisionPool" from "provisionPool-governor". 

Oh, wait. `getVatDetails` might return names like "zcf-b1-db93f-provisionPool" and "zcf-b1-9f877-provisionPool-governor", so you might have to verify with `name.endsWith('provisionPool')`.

File: a3p-integration/proposals/p:upgrade-19/provisionPool.test.js
Comment (Chris-Hibbert) [inline a3p-integration/proposals/p:upgrade-19/provisionPool.test.js]:
```suggestion
      { errorMessage: 'Provision pool not able to swap USDC_axl for IST.' },
```

File: packages/inter-protocol/src/provisionPoolKit.js
Comment (anilhelvaci) [reply packages/inter-protocol/src/provisionPoolKit.js]: resolved in https://github.com/Agoric/agoric-sdk/commit/58e75831a65f90b8259bfdfaaf59b270294a3f6d

**Review by Chris-Hibbert — CHANGES_REQUESTED** [review]
I requested a few clean-ups, but this looks good. Thanks.

File: a3p-integration/proposals/p:upgrade-19/provisionPool.test.js
Comment (anilhelvaci) [reply a3p-integration/proposals/p:upgrade-19/provisionPool.test.js]: > please pull `bankSend`, `provision`, `introduceAndProvision`, `getProvisionedAddress`, and `checkUserProvisioned` out to a separate library file that can be shared. We shouldn't have to define these more than once.

https://github.com/Agoric/agoric-sdk/issues/10514 already tracks this one. I can put those methods into a helper module then move them to the shared library as part of #10514. I think we shouldn't wait for the shared library to land this PR.

File: packages/inter-protocol/src/provisionPoolKit.js
Comment (anilhelvaci) [reply packages/inter-protocol/src/provisionPoolKit.js]: will do

File: a3p-integration/proposals/p:upgrade-19/provisionPool.test.js
Comment (anilhelvaci) [reply a3p-integration/proposals/p:upgrade-19/provisionPool.test.js]: Alright, I'll switch to validating by vat-name instead of vatID 👍

File: packages/inter-protocol/src/provisionPoolKit.js
Comment (anilhelvaci) [reply packages/inter-protocol/src/provisionPoolKit.js]: resolved in https://github.com/Agoric/agoric-sdk/commit/9c757093b0d5492e178362e5458e02e081cbd2b0

File: a3p-integration/proposals/p:upgrade-19/provisionPool.test.js
Comment (anilhelvaci) [reply a3p-integration/proposals/p:upgrade-19/provisionPool.test.js]: resolved in https://github.com/Agoric/agoric-sdk/commit/9c757093b0d5492e178362e5458e02e081cbd2b0

File: a3p-integration/proposals/p:upgrade-19/provisionPool.test.js
Comment (anilhelvaci) [reply a3p-integration/proposals/p:upgrade-19/provisionPool.test.js]: resolved in https://github.com/Agoric/agoric-sdk/commit/9c757093b0d5492e178362e5458e02e081cbd2b0

**Comment by anilhelvaci** [comment]
We talked about commenting about `upgrade-19` stuff in `upgrade.go` however the CI seems to be failing because it expects `upgrade-paRegistry.js` core eval to be executed:

```
registry › priceAuthorityRegistry upgrade

  registry.test.js:16

   15: test('priceAuthorityRegistry upgrade', async t => {
   16:   t.is(await getIncarnation('priceAuthority'), 1); 
   17:                                                    

  Difference (- actual, + expected):

  - 0
  + 1
```

I think the issue is related to this PR that recently landed https://github.com/Agoric/agoric-sdk/pull/10418 @Chris-Hibbert

File: a3p-integration/proposals/p:upgrade-19/provisionPool.test.js
Comment (Chris-Hibbert) [reply a3p-integration/proposals/p:upgrade-19/provisionPool.test.js]: > I think we shouldn't wait for the shared library to land this PR

That's fine.

**Comment by Chris-Hibbert** [comment]
> however the CI seems to be failing because it expects upgrade-paRegistry.js core eval to be executed:

I ran into a similar issue. My conclusion was that the upgrades that are not in U18 should be cleanly moved to u19 and tested there. If that means you have to move some tests from U18 or correct some expected output in U18 in `a3p-integration`, then that's the right approach. Leaving those tests anachronistically in U18 in `a3p-integration` will be more confusing than cleaning it up.

**Comment by anilhelvaci** [comment]
> I ran into a similar issue. My conclusion was that the upgrades that are not in U18 should be cleanly moved to u19 and tested there. If that means you have to move some tests from U18 or correct some expected output in U18 in a3p-integration, then that's the right approach. Leaving those tests anachronistically in U18 in a3p-integration will be more confusing than cleaning it up.

Yep, I think it's resolved in master.

I rebased my branch on top of the master. Could you please approve the PR if it looks good to you? @Chris-Hibbert

File: packages/vats/src/proposals/upgrade-provisionPool-proposal.js
Comment (Chris-Hibbert) [inline packages/vats/src/proposals/upgrade-provisionPool-proposal.js:5]: not a big deal,but we usually call this `trace`.

**Review by Chris-Hibbert — APPROVED** [review]
LGTM

**Pull request merged**
